From bf8a132fba75c1ea7a563c91478466f654e8cfac Mon Sep 17 00:00:00 2001
From: Iain Sandoe <iain@sandoe.co.uk>
Date: Sun, 6 Sep 2020 14:36:47 +0100
Subject: [PATCH] testsuite, Darwin, Arm64 : Amend segaddr test.

The target does not accept no-pie, so we must not try it.

(cherry picked from commit d6af9eadb855a199913e9a1bac664f2528131165)
---
 gcc/testsuite/gcc.dg/darwin-segaddr.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git gcc/testsuite/gcc.dg/darwin-segaddr.c gcc/testsuite/gcc.dg/darwin-segaddr.c
index 526db77bd9c..fcc324b3031 100644
--- gcc/testsuite/gcc.dg/darwin-segaddr.c
+++ gcc/testsuite/gcc.dg/darwin-segaddr.c
@@ -1,7 +1,8 @@
 /* Check that -segaddr gets through and works.  */
 /* { dg-do run { target *-*-darwin* } } */
 /* { dg-options "-O0 -segaddr __TEST 0x200000 -fno-pie" { target { *-*-darwin* && { ! lp64 } } } } */
-/* { dg-options "-O0 -segaddr __TEST 0x110000000 -fno-pie" { target { *-*-darwin* && lp64 } } } */
+/* { dg-options "-O0 -segaddr __TEST 0x110000000 -fno-pie" { target { *-*-darwin[1456789]* && lp64 } } } */
+/* { dg-options "-O0 -segaddr __TEST 0x110000000 " { target { *-*-darwin2* && lp64 } } } */
 
 extern void abort ();
 
-- 
2.40.1

